prometheus/src/main/scala/filodb/prometheus/parse/LegacyParser.scala

```diff
@@ -356,6 +356,7 @@ trait ExpressionParser extends AggregatesParser with SelectorParser with Numeric
 
     "(" ~ expression ~ ")" ^^ {
       case "(" ~ ep ~ ")" => PrecedenceExpression(ep)
+      case _ => ???
```

Comment:
what was missing here ? Can we call explicitly ?

Reply:
This was complaining that `match may not be exhaustive`. So I added this line to catch all other cases compiler things we're missing. Since the block was already working with this setup in the old version, this shouldn't be an issue.